src/representations/VolumeDataRepresentation.tsx

```diff
@@ -1,12 +1,78 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 
 import VolumeRepresentation from '../core/VolumeRepresentation';
 import ImageData from '../core/ImageData';
 import PointData from '../core/PointData';
 import DataArray from '../core/DataArray';
 import VolumeController from '../core/VolumeController';
 
+interface VolumneDataRepresentationProps {
+  /**
+   * The ID used to identify this component.
+   */
+  id?: string;
+  /**
+   * Number of points along x, y, z
+   */
+  dimensions?: number[];
+  /**
+   * Spacing along x, y, z between points in world coordinates
+   */
+  spacing?: number[];
+  /**
+   * World coordinate of the lower left corner of your vtkImageData (i=0, j=0, k=0).
+   */
+  origin?: number[];
+  /**
+   * Use RGB values to attach to the points/vertex
+   */
+  rgb?: number[];
+  /**
+   * Use RGBA values to attach to the points/vertex
+   */
+  rgba?: number[];
+  /**
+   * Field values to attach to the points
+   */
+  scalars?: number[];
+  /**
+   * Types of numbers provided in scalars
+   */
+  scalarsType?: string;
+  /**
+   * Properties to set to the mapper
+   */
+  mapper?: object;
```

Comment:
Do these objects map directly to classes like `vtkMapper`, `vtkVolume`, `vtkProperty`, etc...? Could the object eventually be converted to the types on those classes?

Reply:
I'm not sure, they were defined just as Object on prop-types 😔

Reply:
These are objects that represent the settable properties for the mapper/property/etc. We can feasibly use the initial values types for the object constructors here. For instance, here is the [vtkMapper initial values interface](https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/Mapper/index.d.ts#L26).